filters/date.go

```diff
@@ -1,58 +1,145 @@
 package filters
 
 import (
+	"strings"
 	"time"
 
 	"github.com/acstech/go-strftime"
 	"github.com/acstech/liquid/core"
 )
 
 var (
-	zeroTime = time.Time{}
+	zeroTime         = time.Time{}
+	timeInputFormats = []string{
+		"2006-01-02 15:04:05 -0700",
+		"2006-01-02 15:04:05",
+		"2006-01-02",
+		"2006-01-02T15:04:05",
+		"2006-01-02T15:04:05Z",
+		"1/2/2006 03:04:05 PM",
+		"1/2/2006 3:04:05 PM",
+		"1/2/2006",
+		"1/2/2006 15:04:05",
+	}
 )
 
-// Creates an date filter
+// DateFactory creates an date filter
 func DateFactory(parameters []core.Value) core.Filter {
-	if len(parameters) == 0 {
+	switch len(parameters) {
+	case 1:
+		return (&DateFilter{format: parameters[0]}).ToString
+	case 2:
+		switch parameters[1].(type) {
+		case *core.StaticIntValue, *core.StaticFloatValue:
+			return (&DateFilter{format: parameters[0], offset: parameters[1]}).ToString
+		}
+
+		return (&DateFilter{format: parameters[0], inputFormat: parameters[1]}).ToString
+
+	case 3:
+		switch parameters[1].(type) {
+		case *core.StaticIntValue, *core.StaticFloatValue:
+			return (&DateFilter{format: parameters[0], offset: parameters[1], inputFormat: parameters[2]}).ToString
+		}
+
+		return (&DateFilter{format: parameters[0], inputFormat: parameters[1], offset: parameters[2]}).ToString
+
+	default:
```

Comment:
I would argue that additional parameters after 3 should just be ignored if for some reason someone accidentally passes something additional in, and that the noop should only be when there are no parameters.